amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/declaration/SchemaVersion.scala

```diff
@@ -4,33 +4,44 @@ import amf.client.render.{JSONSchemaVersions => ClientJSONSchemaVersions, JSONSc
 import amf.core.errorhandling.ErrorHandler
 import amf.plugins.features.validation.CoreValidations
 
-abstract class JSONSchemaVersion(val name: String)
+abstract class SchemaVersion(val name: String)
+
+object SchemaVersion {
+  def fromClientOptions(schema: ClientJSONSchemaVersion): SchemaVersion = schema match {
+    case ClientJSONSchemaVersions.DRAFT_07 => JSONSchemaDraft7SchemaVersion
+    case _                                 => JSONSchemaDraft4SchemaVersion
+  }
+}
+
+abstract class RAMLSchemaVersion(override val name: String) extends SchemaVersion(name)
```

Comment:
[Out of scope of this story] Why do we version RAML & OAS Schemas? Not clear why de we need to synthesize a RAML/OAS schema version? After this doing this story, did you see any use for these?

Reply:
OAS versions are primarily used due to a lack of correct design in the OasTypeParser. OAS Schema Objects have some differences with Json Schema ones, primarily in validation. That is why we version them.
RAML version was used as a config setting to parse Raml Types in Async.

Ideally, we would only need them to know which TypeParser to build (this mostly applies to OAS and JsonSchema). They should be ValueObjects with no other responsibility than identifying the parser I wish to build and parse with.